curvefs/src/client/metaserver_client.cpp

```diff
@@ -17,65 +17,301 @@
 
 /*
  * Project: curve
- * Created Date: Thur May 27 2021
- * Author: xuchaojie
+ * Created Date: Thur Jun 15 2021
+ * Author: lixiaocui
  */
 
 #include "curvefs/src/client/metaserver_client.h"
 
 namespace curvefs {
 namespace client {
 
-CURVEFS_ERROR MetaServerClientImpl::GetDentry(uint32_t fsId, uint64_t inodeid,
-              const std::string &name, Dentry *out) {
-    return CURVEFS_ERROR::OK;
+CURVEFS_ERROR
+MetaServerClientImpl::MetaServerRPCExcutor::DoRPCTask(RPCFunc task) {
+    brpc::Channel channel;
+    int ret = channel.Init(opt_.msaddr.c_str(), nullptr);
+    if (ret != 0) {
+        LOG(WARNING) << "Init channel failed, addr = " << opt_.msaddr;
+        return static_cast<CURVEFS_ERROR>(-EHOSTDOWN);
+    }
+
+    brpc::Controller cntl;
+    cntl.set_timeout_ms(opt_.rpcTimeoutMs);
+
+    ret = task(&channel, &cntl);
+    return static_cast<CURVEFS_ERROR>(ret);
 }
 
-CURVEFS_ERROR MetaServerClientImpl::ListDentry(uint32_t fsId, uint64_t inodeid,
-        const std::string &last, uint32_t count) {
+CURVEFS_ERROR
+MetaServerClientImpl::Init(const MetaServerOption &metaopt,
+                           MetaServerBaseClient *baseclient) {
+    basecli_ = baseclient;
+    excutor_.SetOption(metaopt);
     return CURVEFS_ERROR::OK;
 }
 
-CURVEFS_ERROR MetaServerClientImpl::UpdateDentry(const Dentry &dentry) {
-    return CURVEFS_ERROR::OK;
+
+#define RPCTaskDefine                                                          \
+    [&](brpc::Channel * channel, brpc::Controller * cntl) -> int
+
+CURVEFS_ERROR MetaServerClientImpl::GetDentry(uint32_t fsId, uint64_t inodeid,
+                                              const std::string &name,
+                                              Dentry *out) {
+    auto task = RPCTaskDefine {
+        GetDentryResponse response;
+        basecli_->GetDentry(fsId, inodeid, name, &response, cntl, channel);
+
+        if (cntl->Failed()) {
+            LOG(WARNING) << "GetDentry Failed, errorcode = "
+                         << cntl->ErrorCode()
+                         << ", error content:" << cntl->ErrorText()
+                         << ", log id = " << cntl->log_id();
+            return -cntl->ErrorCode();
+        }
+
+        CURVEFS_ERROR retcode = CURVEFS_ERROR::FAILED;
+        MetaStatusCode stcode = response.statuscode();
+        MetaServerStatusCode2CurveFSErr(stcode, &retcode);
+        LOG_IF(WARNING, retcode != CURVEFS_ERROR::OK)
+            << "GetDentry: fsId = " << fsId << ", inodeid = " << inodeid
+            << ", name = " << name << ", errcode = " << retcode
+            << ", errmsg = " << MetaStatusCode_Name(stcode);
+
+        if (response.has_dentry()) {
```

Comment:
Do not rely on has_dentry to determine whether the execution is successful. First judge the return value.

The following function is the same

Reply:
above the code has judgement:         LOG_IF(WARNING, retcode != CURVEFS_ERROR::OK). 
here just assign dentry